apps/dashboard/src/components/variable/components/reorder-filter-item.tsx

```diff
@@ -1,80 +1,182 @@
+  const itemRef = useRef<HTMLDivElement>(null);
 
   const filterDef = liquidFilters.find((t) => t.value === value.value);
+  const hasParams = filterDef?.hasParam && filterDef.params;
+
+  const options = useMemo(() => {
+    // if it's digest step events variable then fill the options with the payload variables
+    if (variableName.match(/^steps\..+\.events$/)) {
+      return variables.filter((v) => v.name.startsWith('payload')).map((v) => ({ label: v.name, value: v.name }));
+    }
+
+    return [];
+  }, [variableName, variables]);
 
   return (
     <Reorder.Item
+      ref={itemRef}
       value={value}
       className="group mb-0 flex items-center gap-1.5 rounded-md p-0.5"
       whileDrag={{ scale: 1.02 }}
       transition={{
+        type: 'keyframes',
         duration: 0.15,
         ease: [0.32, 0.72, 0, 1],
       }}
+      style={{ x }}
+      dragListener={false}
+      dragControls={controls}
+      onDragStart={() => {
+        if (itemRef.current) {
+          const height = itemRef.current.getBoundingClientRect().height;
+          itemRef.current.style.minHeight = `${height}px`;
+        }
+      }}
+      onDragEnd={() => {
+        // reset the x position to 0 to avoid the item from being dragged out of the container
+        x.set(0);
+
+        if (itemRef.current) {
+          itemRef.current.style.minHeight = '';
+        }
+      }}
       {...rest}
     >
-      <GripVertical className="text-text-soft h-3.5 w-3.5" />
-      <div className="flex flex-1 items-center gap-1">
-        <div className="border-stroke-soft text-text-sub text-paragraph-xs bg-bg-weak rounded-8 flex w-full flex-row items-center border">
-          <Tooltip delayDuration={700}>
-            <TooltipTrigger asChild>
-              <div className="cursor-help px-2 py-1.5 pr-0">{filterDef?.label}</div>
-            </TooltipTrigger>
-            <TooltipContent side="left" align="center" className="font-mono text-[10px]">
-              {filterDef?.example}
-            </TooltipContent>
-          </Tooltip>
-          {filterDef?.hasParam && filterDef.params && (
-            <div className="flex flex-1 flex-col gap-1 py-1">
-              {filterDef.params.map((param, paramIndex) => (
-                <InputPure
-                  key={paramIndex}
-                  value={value.params?.[paramIndex] || ''}
-                  onChange={(e) => {
-                    const newParams = [...(value.params || [])];
-                    newParams[paramIndex] = e.target.value;
-                    onParamChange(index, newParams);
-                  }}
-                  className="border-stroke-soft ml-1 h-[20px] w-[calc(100%-8px)] border-l pl-1 text-xs"
-                  placeholder={param.placeholder}
-                  title={param.description}
+      <Accordion className={'bg-bg-weak flex w-full flex-col gap-2 rounded-lg'} type="single" collapsible>
```

Comment:
all the below are the improvements coming from the latest UI from Figma

Reply:
<img width="348" alt="image" src="https://github.com/user-attachments/assets/6c56ad2c-3e1b-4f07-8794-7123408e7f23" />
for now it's fine but we will need to update the filter UI as currently it is not clear that it's an accordion and users need to click it

Reply:
I agree, I'll remove the accordion because it creates more problems than helps.